tests/util/integration.go

```diff
@@ -145,17 +151,53 @@ func K3sStartServer(inputArgs ...string) (*K3sServer, error) {
 
 	k3sCmd := append([]string{"server"}, cmdArgs...)
 	cmd := exec.Command(k3sBin, k3sCmd...)
+	// Give the server a new group id so we can kill it and its children later
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 	cmdOut, _ := cmd.StderrPipe()
 	cmd.Stderr = os.Stderr
 	err = cmd.Start()
 	return &K3sServer{cmd, bufio.NewScanner(cmdOut), k3sLock}, err
 }
 
-// K3sKillServer terminates the running K3s server and unlocks the file for
-// other tests
-func K3sKillServer(server *K3sServer) error {
-	if err := server.cmd.Process.Kill(); err != nil {
+// K3sKillServer terminates the running K3s server and its children
+// and unlocks the file for other tests
+func K3sKillServer(server *K3sServer, releaseLock bool) error {
+	pgid, err := syscall.Getpgid(server.cmd.Process.Pid)
+	if err != nil {
+		return err
+	}
+	if err := syscall.Kill(-pgid, syscall.SIGKILL); err != nil {
+		return err
+	}
+	if releaseLock {
+		return flock.Release(server.lock)
+	}
+	return nil
+}
+
+// K3sCleanup attempts to cleanup networking and files leftover from an integration test
+func K3sCleanup(server *K3sServer, releaseLock bool) error {
```

Comment:
This seems to share a lot of the logic with the k3s-killall.sh script; would it be easier to just call that?

Reply:
So we don't have the k3s-killall.sh script available to these tests, because we only generate k3s-killall.sh when installing K3s from the https://get.k3s.io.

Reply:
Maybe we should stop generating k3s-killall.sh and k3s-uninstall.sh dynamically and just host them when pulling k3s? Or provide a way for local dev to call `make kill` and have the install.sh just generate the k3s-killall.sh for us locally.

Reply:
Yeah, it has been requested as a release artifact in the past, so that would probably be appreciated.